### PR TITLE
Half-norm^Half Proximal Operator + standard prox

### PR DESCRIPTION
**Pull request by rjbaraldi**
Modeled after ProximalOperator NormL0. 
Put in a very basic test that we can certainly change. 
Also open to name changes.

### CODE REVIEW

File: src/normLhalf.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  return f.lambda * sum(sqrt.(abs.(y)))
+end
+
+fun_name(f::RootNormLhalf) = "L_(1/2)^(1/2) pseudo-norm"
+fun_dom(f::RootNormLhalf) = "AbstractArray{Real}, AbstractArray{Complex}"
+fun_expr(f::RootNormLhalf{T}) where {T <: Real} = "x ↦ (λ/2)||x||^(1/2)_(1/2)"
+fun_params(f::RootNormLhalf{T}) where {T <: Real} = "λ = $(f.lambda)"
+
+function prox_naive(f::RootNormLhalf, x::AbstractArray{T}, gamma::Real=1) where {R, T <: Real}
```
Comment (dpo) [inline src/normLhalf.jl:50]: why is this called naive?

File: src/normLhalf.jl [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+"""
+**``L_1/2^(1/2)`` pseudo-norm**
+    RootNormLhalf(λ=1)
+Returns the function
+```math
+f(x) = λ\\cdot \\sum |x|\\^{1/2}
+```
+for a nonnegative parameter `λ`.
+"""
+struct RootNormLhalf{R <: Real} <: ProximableFunction
```
Comment (dpo) [inline src/normLhalf.jl:14]: there are no shifts here right? This should go to ProximalOperators then.

File: src/normLhalf.jl (hunk 1 above)
Comment (rjbaraldi) [reply src/normLhalf.jl:50]: It was the naming convention in ProximalOperators for some reason. Most of their functions seem to have it? It's never used so we can exclude it.

File: src/normLhalf.jl (hunk 2 above)
Comment (rjbaraldi) [reply src/normLhalf.jl:14]: Yep that's fine with me if one of use wants to send a PR or something? Not sure how that works or what the etiquette is.

Comment (rjbaraldi) [reply src/normLhalf.jl:14]: I'm shy though. Should I send an email or just github?

Comment (dpo) [reply src/normLhalf.jl:14]: ok, let's leave it here for now

**Review by dpo — APPROVED** [review]
Looks good to me! Thanks!

**Comment by dpo** [comment]
I think this is ready to go, if you want to merge.

File: src/normLhalf.jl (hunk 2 above)
Comment (rjbaraldi) [reply src/normLhalf.jl:14]: hahaha alright i guess i can add it once it works

**Pull request merged**